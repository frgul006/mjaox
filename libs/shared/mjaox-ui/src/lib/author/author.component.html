<mat-card class="mjaox-ui-author">
  <mat-card-header>
    <mat-card-title>{{
      author?.firstname + ' ' + author?.lastname
    }}</mat-card-title>
    <mat-card-subtitle>
      <a
        [href]="
          'https://twitter.com/' + author?.twitter + '?ref_src=twsrc%5Etfw'
        "
        class="twitter-follow-button"
        data-show-count="false"
        >Follow {{ author?.twitter }}</a
      ><script
        async
        src="https://platform.twitter.com/widgets.js"
        charset="utf-8"
      ></script>
    </mat-card-subtitle>
  </mat-card-header>
  <img
    *ngIf="author?.profileimage?.handle"
    mat-card-image
    [src]="
      'https://media.graphcms.com/resize=w:500/quality=value:70/compress/' +
      author?.profileimage?.handle
    "
    [alt]="'Photo of ' + author?.firstname + ' ' + author?.lastname"
  />
  <mat-card-content>
    <p>{{ author?.description }}</p>
  </mat-card-content>
  <mat-card-actions>
    <button mat-button>LIKE</button> <button mat-button>SHARE</button>
  </mat-card-actions>
</mat-card>
