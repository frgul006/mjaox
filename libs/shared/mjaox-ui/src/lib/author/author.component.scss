.mjaox-ui-author {
  &__avatar {
    overflow: hidden;
  }
}
