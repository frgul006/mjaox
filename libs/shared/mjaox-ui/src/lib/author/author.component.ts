import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { Author } from '@mjaox/shared/mjaox-model';

@Component({
  selector: 'mjaox-ui-author',
  templateUrl: './author.component.html',
  styleUrls: ['./author.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AuthorComponent {
  @Input() author: Author;
}
