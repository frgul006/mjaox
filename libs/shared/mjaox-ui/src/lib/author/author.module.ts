import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatCardModule } from '@angular/material/card';
import { AuthorComponent } from './author.component';

const COMPONENTS = [AuthorComponent];

@NgModule({
  imports: [CommonModule, MatCardModule, MatButtonModule],
  declarations: [...COMPONENTS],
  exports: [...COMPONENTS]
})
export class SharedMjaoxUiAuthorModule {}
