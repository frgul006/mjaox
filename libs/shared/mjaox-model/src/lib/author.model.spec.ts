import { Author } from './author.model';

describe('Author Model', () => {
  it('should initiate with defaults', () => {
    const author = new Author();
    expect(author).toMatchSnapshot();
  });
});
