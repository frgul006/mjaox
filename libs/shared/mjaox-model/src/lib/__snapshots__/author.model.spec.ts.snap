// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Model should initiate with defaults 1`] = `Author {}`;
